# Q1 : 
   A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [1]:
import pandas as pd 
import numpy  as np
from scipy import stats
from scipy.stats import norm

In [2]:
#Loading Dataset

df1 = pd.read_csv("D:\ExcelR\Data Science\Assignments\Hypothesis Testing\Cutlets.csv")
df1.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

-  Assumption :- 
   -  Null Hypothesis as Ho :  Unit A = Unit B and
   -  Alternate Hypothesis as Ha : Unit A ≠ Unit B  

In [3]:
unitA = pd.Series(df1.iloc[:,0])
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [4]:
unitB = pd.Series(df1.iloc[:,1])
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [5]:
# 2-sample 2-tail t test

p_value = stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [6]:
p_value[1]

0.4722394724599501

In [7]:
#Compairing P_value with α = 0.05 (at 5% significance level)

As (p_value = 0.472) > (α = 0.05) accept null hypothesis i.e. μ1=μ2. There is no differnce between in diameter of cutlets between two units.

# Q2 :
A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [8]:
#Loading Dataset

df2 = pd.read_csv("D:\ExcelR\Data Science\Assignments\Hypothesis Testing\LabTAT.csv")
df2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

Assumptions :-
-  Null Hypothesis as Ho : Lab1 = Lab2 = Lab3 = Lab4
-  Alternate Hypothesis as Ha : Atleast one of them is different.

In [9]:
p_value = stats.f_oneway(df2.iloc[:,0],df2.iloc[:,1],df2.iloc[:,2],df2.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [10]:
p_value[1]

2.1156708949992414e-57

In [11]:
#Compairing P_value with α = 0.05 (at 5% significance level)

As (p_value=0) < (α=0.05);
Reject null hypothesis i.e. atleast one sample TAT population mean is different thus there is difference in average in avarage TAT of reports of the laboratories on their preferred list.

# Q3
Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [13]:
#Loading Dataset

df3 = pd.read_csv("D:\ExcelR\Data Science\Assignments\Hypothesis Testing\BuyerRatio.csv")
df3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Assumptions :-
-  Null Hypothesis as Ho : Male-Female buyer raitos are similar across region (Independence of categorical variables)
-  Alternate hypothseis as Ha : Male-Female buyer ratios are NOT similar across region (Dependence of categorical variables)

In [14]:
df3_table = df3.iloc[:,1:6]
df3_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [15]:
df3_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [17]:
val = stats.chi2_contingency(df3_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [18]:
type(val)

tuple

In [20]:
no_of_rows=len(df3_table.iloc[0:2,0])
no_of_columns=len(df3_table.iloc[0,0:4])
degree_of_f=(no_of_rows-1)*(no_of_columns-1)
print('Degree of freedom=',degree_of_f)

Degree of freedom= 3


In [21]:
Expected_value = val[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [22]:
from scipy.stats import chi2

In [23]:
chi_square=sum([(o-e)**2/e for o,e in zip(df3_table.values,Expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [24]:
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

In [25]:
if chi_square_statestic >= critical_value:
    print('Dependent (reject H0)')
else:
    print('Independent (fail to reject H0)')

Independent (fail to reject H0)


In [26]:
pvalue=1-chi2.cdf(chi_square_statestic,3)
pvalue

0.6787446296467897

In [27]:
if pvalue <= 0.05:
    print('Dependent (reject H0)')
else:
    print('Independent (fail to reject H0)')

Independent (fail to reject H0)


# Q4
TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

In [28]:
#Loading Dataset

df4 = pd.read_csv("D:\ExcelR\Data Science\Assignments\Hypothesis Testing\Costomer+OrderForm.csv")
df4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [30]:
df4.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [32]:
df4.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [33]:
df4.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [34]:
df4.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [35]:
observed = ([[271,267,269,280],[29,33,31,20]])
observed

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [36]:
from scipy.stats import chi2_contingency

In [37]:
stat, p, dof, expected = chi2_contingency([[271,267,269,280],[29,33,31,20]])
stat

3.858960685820355

In [38]:
p

0.2771020991233135

In [41]:
print('dof=%d' % dof)
print(expected)

dof=3
[[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [43]:
alpha = 0.05
prob = 1-alpha
critical = chi2.ppf(prob, dof)
print('probability=%.3f, critical=%.3f, stat=%.3f' % (prob, critical, stat))
if abs(stat) >= critical:
    print('Dependent (reject Ho), varables are related')
else:
    print('Independent (fail to reject Ho), varables are not related')

probability=0.950, critical=7.815, stat=3.859
Independent (fail to reject Ho), varables are not related


In [44]:
print('significance=%.3f, p=%.3f' % (alpha, p))
if p <= alpha:
    print('Dependent (rejecct Ho)')
else:
    print('Independent (fail to reject Ho)')

significance=0.050, p=0.277
Independent (fail to reject Ho)
